clang/lib/AST/Interp/Interp.h

```diff
@@ -1005,12 +1008,23 @@ bool SetThisField(InterpState &S, CodePtr OpPC, uint32_t I) {
 template <PrimType Name, class T = typename PrimConv<Name>::T>
 bool GetGlobal(InterpState &S, CodePtr OpPC, uint32_t I) {
   const Block *B = S.P.getGlobal(I);
+
+  if (!CheckConstant(S, OpPC, B->getDescriptor()))
+    return false;
   if (B->isExtern())
     return false;
   S.Stk.push<T>(B->deref<T>());
   return true;
 }
 
+/// Same as GetGlobal, but without the checks.
+template <PrimType Name, class T = typename PrimConv<Name>::T>
+bool GetGlobalUnchecked(InterpState &S, CodePtr OpPC, uint32_t I) {
+  auto *B = S.P.getGlobal(I);
+  S.Stk.push<T>(B->deref<T>());
+  return true;
+}
+
```

Comment:
Would it be reasonable to implement `GetGlobal` in terms of `GetGlobalUnchecked` so we can share code, or is that not feasible because `GetGlobalUnchecked` will push something to the stack that the checks may then reject?

Reply:
It's a bit involved because `GetGlobal` can't just call `GetGlobalUnchecked` since they would both have to call `S.P.getGlobal(I)`... I mean, that's not a problem per se but a little ugly and slower than it needs to be. So we'd have to use a third function that just does `S.Stk.push<T>(B->deref<T>()); return true;`.

Reply:
Eh probably not worth it then.